doc/rst/technotes/firstClassProcedures.rst

```diff
@@ -0,0 +1,193 @@
+
+Notice that the formals of the procedure type have specified names,
+``x`` and ``y``. Currently, formal names participate in typing such
+that a procedure defined with formals named ``x`` and ``y`` cannot be
+assigned to a procedure defined with different formal names. In the
+below example, assignment of two procedure values will fail because
+the formal names are different:
+
+.. code-block:: chapel
+
+  proc foo(x: int, y: int): int { return x + y; }
+  proc bar(a: int, b: int): int { return a + b; }
+  writeln(foo.type:string);   // 'proc(x: int, y: int): int'
+  var x = foo;                // ^ (same type) ^
+  writeln(bar.type:string);   // 'proc(a: int, b: int): int'
+  x = bar;                    // Error!
+
+There are scenarios where a user may want to pass around procedures 
+without regard for their formal names. Anonymous ``_`` formals are a
+feature under active development that are intended to support this
+behavior.
+
+New Syntax for Constructing Anonymous Procedures 
+------------------------------------------------
+
+A new syntax for constructing anonymous procedures has been introduced
+which more closely mirrors traditional procedure definition.
+
+.. code-block:: chapel
+
+  // Define a procedure named 'foo'.
+  proc foo(x: int, y: int): int { return x + y; }
+
+  // Define an anonymous procedure bound to the constant variable 'bar'.
+  const bar = proc(a: int, b: int): int { return a + b; };
+
+The Formals of Procedure Types May Be Anonymous
+-----------------------------------------------
+
+.. note::
+
+  This section describes unstable functionality that may not be implemented.
+
+A formal in a procedure type may be anonymous. This may be done by
+naming the formal ``_``, similar to what is written to discard
+tuple elements when de-tupling.
+
+An anonymous formal in a procedure type indicates that values of this
+type may have formals with any name. The formal name no longer plays
+a significant role in typing.
+
+.. code-block:: chapel
+
+  proc foo(x: int, y: int): int { return x + y; }
+  proc bar(a: int, b: int): int { return a + b; }
+
+  // Here the formals of 'T' are anonymous.
+  type T = proc(_: int, _: int): int;
+
+  var x: T = foo;             // OK, T's formals are anonymous '_'.
+  writeln(foo.type:string);   // 'proc(x: int, y: int): int'
+  writeln(x.type:string);     // 'proc(_: int, _: int): int'
+  writeln(bar.type:string);   // 'proc(a: int, b: int): int'
+  x = bar;                    // OK!
+
+In the above example, two procedures are declared with different formal
+names, but otherwise identical types. A local variable `x` is declared
+with a procedure type that has anonymous formals. This enables the
+procedures `foo` and `bar` to be assigned freely to `x`.
+
+.. note::
+
+  Currently, it is not possible for procedure definitions to declare
+  anonymous formals. It has been indicated that such a feature might
+  be useful, so it may be added in the future.
```

Comment:
How about this:

Currently anonymous formals are not implemented.
This feature may be added in the future.